src/router/storeSync/storeSync.routerPlugin.js

```diff
@@ -39,7 +39,8 @@ const isRoutePushNeeded = (store, currentRoute) => {
 
 // flag to distinguish URL change originated by this module or by another source
 let routeChangeIsTriggeredByThisModule = false
-
+// flag to tell us if we need to enforce a query redirection when coming from the legacy router
+let needRouteChangeBecauseOfLegacy = true
```

Comment:
I'm not sure it's a good idea to set it to `true` by default.
Maybe it should be set to `true` the first time we enter the LegacyView. If I understand correctly the goal here is to make sure we exit this view, so it would make more sense this way (here I fear it will trigger something unwanted when entering the normal MapView...)

Reply:
That would make more sense. Functionally there is no real issue (the map view will go through the router one additional time, but it won't crash or go infinite), but it would be cleaner to only do it when we went through the legacy router first.